### ISSUE TITLE
HPC-X rebuild script doesn't work to create new init file for OMPI and UCX

### ISSUE THREAD

**Issue by dmitrygx** [openucx/ucx#8837]
Hi,

I apologise for submitting the issue here, because it is not directly related to UCX, but didn't find a proper way to submit a bug report to HPC-X product.
There is a problem in `hpcx_rebuild.sh` script. It should use `${HPCX_DIR}` instead of `HPCX_DIR`.

Here is a diff how the problem could be probably fixed:
```
--- ./hpcx_rebuild_orig.sh   2023-05-11 14:28:09.806753414 +0300
+++ ./hpcx_rebuild.sh  2023-05-11 17:54:33.021136380 +0300
@@ -294,16 +294,16 @@
 echo "INFO: Setting up new init script and module file..."
 name=$(basename "${ompi_prefix}")
 new_init="${name}.sh"
-new_mod_file="modulefiles/$name"
+new_mod_file="modulefiles/${name}"
 cd "${HPCX_ROOT}"
 cp ${base_init_script} "${new_init}"
 cp modulefiles/${base_module_file} "${new_mod_file}"

-sed -i "s;HPCX_DIR/ompi;HPCX_DIR/$name;" "${new_init}"
+sed -i "s;{HPCX_DIR}/ompi;{HPCX_DIR}/$name;" "${new_init}"
 sed -i "s;hpcx_dir/ompi;hpcx_dir/$name;" "${new_mod_file}"

 if [ "${rebuild_ucx}" = "yes" ]; then
-    sed -i "s;HPCX_DIR/ucx;HPCX_DIR/ucx/$name;" "${new_init}"
+    sed -i "s;{HPCX_DIR}/ucx;{HPCX_DIR}/ucx/$name;" "${new_init}"
     sed -i "s;hpcx_dir/ucx;hpcx_dir/ucx/$name;" "${new_mod_file}"
 fi
 echo "INFO: Done"
```

**Comment by dmitrygx** [comment]
The issue is still available in HPC-X v2.15 that was released on May 08, 2023.

**Comment by artemry-nv** [comment]
Thanks for reporting!
Fixed, will be available in HPC-X 2.16 (Jul).

**Comment by dmitrygx** [comment]
> Thanks for reporting! Fixed, will be available in HPC-X 2.16 (Jul).

@artemry-nv Thanks for the support! Everything works fine in all new versions.
